src/services/ecommerce/payments/hooks.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery, useQueryClient, useMutation } from "@tanstack/react-query";
+import { paymentsApi } from "../client";
+import type {
+  PaymentsApiPaymentsBasketsListRequest,
+  PaymentsApiPaymentsBasketsCreateFromProductCreateRequest,
+  PaymentsApiPaymentsBasketsAddDiscountCreateRequest,
+  PaymentsApiPaymentsCheckoutCreateRequest,
+} from "../generated/v0/api";
+
+const usePaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+) =>
+  useQuery({
+    queryKey: ["paymentsBaskets"],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsList(options);
+      return response.data;
+    },
+  });
+
+const useDeferredPaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+  enabled: boolean,
+) =>
+  useQuery({
+    queryKey: ["paymentsBaskets"],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsList(options);
+      return response.data;
+    },
+    enabled: enabled,
+  });
+
+const usePaymentsBasketRetrieve = (id: number) => {
+  return useQuery({
+    queryKey: ["paymentsBaskets", id],
+    queryFn: async () => {
+      const response = await paymentsApi.paymentsBasketsRetrieve({ id });
+      return response.data;
+    },
+  });
+};
+
+const useDeferredPaymentsBasketRetrieve = (id: number, enabled: boolean) => {
```

Comment:
I would get rid of this and make the one above 

```
import type { UseQueryOptions } from "@tanstack/react-query"
const usePaymentsBasketRetrieve = (id: number, opts: opts: Omit<UseQueryOptions, "queryKey"> = {}) => {
 return useQuery({
    queryKey,
    queryFn,
    ...opts 
  })
}
```
and just pass the opts where you need them.